public/js/draw/draw.js

```diff
@@ -389,3 +390,98 @@ function regionClicked(e) {
         prevY = position.y;
     }
 }
+
+
+/**
+ * convertSvgToJson()
+ * Parse SVG elements into a JSON list.
+ */
+function convertSvgToJson(gId) {
+    'use strict';
+
+    var texts = [];
+    var shapes = [];
+    var paths = [];
+    var svgElements = document.getElementById('mySVG').children;
+    for (var i = 0; i < svgElements.length; i++) {
+        if (svgElements[i].tagName == "g" && svgElements[i].id != "regions") {
+            var textObj = "";
+            var gText = svgElements[i].querySelector("text");
+            if (gText !== null) {
+                textObj = {
+                    "graph"     : parseInt(gId),
+                    "rId"       : "",
+                    "pos"       : [ parseFloat(gText.getAttribute("x")),
+                                    parseFloat(gText.getAttribute("y"))
+                                    ],
+                    "text"      : gText.textContent,
+                    "align"     : "begin",
+                    "fill"      : ""
+                };
+               texts.push(textObj); 
+            }
+
+            var gRect = svgElements[i].querySelector("rect");
+            shapes.push({
+                    "graph"     : parseInt(gId),
+                    "id_"       : "",
+                    "pos"       : [ parseFloat(gRect.getAttribute("x")),
+                                    parseFloat(gRect.getAttribute("y"))
+                                    ],
+                    "width"     : parseFloat(gRect.getAttribute("width")),           
+                    "height"    : parseFloat(gRect.getAttribute("height")),
+                    "fill"      : "none",
+                    "stroke"    : "",
+                    "text"      : [textObj],
+                    "tolerance" : 9,
+                    "type_"     : "Node"                         
+            });
+        } else if (svgElements[i].tagName == "path") {
+            var pathCoords = getPathCoords(svgElements[i].getAttribute("d"));
+            paths.push({
+                    "graph"     : parseInt(gId),
+                    "id_"       : "",
+                    "points"    : pathCoords,
+                    "fill"      :"none",
+                    "stroke"    : "",
+                    "isRegion"  : false,
+                    "source"    : getClosestText(pathCoords[0], shapes),
+                    "target"    : getClosestText(pathCoords[1], shapes)
+            });
+        }
+    }
+    return JSON.stringify({texts:texts, shapes:shapes, paths:paths});
+}
+
+
+/**
+ * getPathCoords(pathCoords)
```

Comment:
Putting the function name in the JSdoc is not only redundant, but also bad from a maintainability and consistency standpoint. When the function is renamed in the future, the JSdoc might not be changed.